### PR TITLE
Provide expected outputs

### LINKED ISSUE #353: Provide expected output in errors, where there is none
https://github.com/foundry-rs/starknet-foundry/pull/340#discussion_r1279273268

### PR DESCRIPTION
**Pull request by drknzz**
Closes #353 

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cast/tests/e2e/declare.rs [hunk 1]
```diff
@@ -85,12 +87,10 @@ async fn wrong_contract_name_passed() {
...
         .args(args);
 
-    let output = String::from_utf8(snapbox.assert().failure().get_output().stderr.clone()).unwrap();
-
-    assert!(
-        output.contains("Failed to find contract nonexistent in starknet_artifacts.json"),
-        "Expected error message not found in stderr: {output}",
-    );
+    snapbox.assert().failure().stderr_matches(indoc! {r#"
+        [..]Failed to find contract nonexistent in starknet_artifacts.json[..]
```
Comment (THenry14) [inline crates/cast/tests/e2e/declare.rs:91]: should this also include an information about command, to be in line with other outputs?

File: crates/cast/tests/e2e/declare.rs [hunk 2]
```diff
@@ -173,13 +172,9 @@ fn scarb_no_artifacts(contract_path: &str, accounts_file_path: &str) {
...
-
-    assert!(
-        stderr_output.contains(
-            "is set to 'true' under your [[target.starknet-contract]] field in Scarb.toml"
-        ),
-        "Expected error message not found in stderr: {stderr_output}",
-    );
+
+    snapbox.assert().failure().stderr_matches(indoc! {r#"
+        [..]is set to 'true' under your [[target.starknet-contract]] field in Scarb.toml[..]
```
Comment (THenry14) [inline crates/cast/tests/e2e/declare.rs:177]: should this also include an information about command, to be in line with other outputs?

File: crates/cast/tests/e2e/multicall/run.rs [hunk 3]
```diff
@@ -105,10 +106,8 @@ async fn test_deploy_success_invoke_fails() {
...
 
     let snapbox = runner(&args);
-    let bdg = snapbox.assert();
-    let out = bdg.get_output();
-
-    let stderr_str =
-        std::str::from_utf8(&out.stderr).expect("failed to convert command output to string");
-    assert!(stderr_str.contains("error: There is no contract at the specified address"));
+    snapbox.assert().stderr_matches(indoc! {r#"
+        ...
```
Comment (THenry14) [inline crates/cast/tests/e2e/multicall/run.rs:110]: the same as previous comments

File: crates/cast/tests/e2e/declare.rs (hunk 1 above)
Comment (drknzz) [reply crates/cast/tests/e2e/declare.rs:91]: Will change it after the #513 is merged

**Review by war-in — APPROVED** [review]

**Pull request merged**